File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        
        if (list1 == null) {
            return list2;
        }
        if (list2 == null) {
            return list1;
        }
        
        ListNode result;
        ListNode rpointer;
        
        if (list1.val < list2.val) {
            result = new ListNode(list1.val);
            
            list1 = list1.next;
        }
        
        else {
            result = new ListNode(list2.val);
            
            list2 = list2.next;
        }
        
        rpointer = result;
        

        
        while (list1 != null || list2 != null) {
            if (list1 == null) {
                rpointer.next = new ListNode(list2.val);
                rpointer = rpointer.next;
                
                list2 = list2.next; 
            }
            else if (list2 == null) {
                rpointer.next = new ListNode(list1.val);
                rpointer = rpointer.next;
                
                list1 = list1.next;     
            }
            
            else if (list1.val <= list2.val) {
                rpointer.next = new ListNode(list1.val);
                rpointer = rpointer.next;
                
                list1 = list1.next;
            }
            else if (list2.val <= list1.val) {
                rpointer.next = new ListNode(list2.val);
                rpointer = rpointer.next;
                
                list2 = list2.next;
            }
            
        }
        
        return result;
        
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        
        
        if list1 is None:
            return list2
        elif list2 is None:
            return list1
        
        sortedHead = ListNode()
        current = sortedHead
        
        while list1 or list2:
            if not list1:
                current.next = list2
                return sortedHead.next
            elif not list2:
                current.next = list1
                return sortedHead.next
            else:
                if list1.val <= list2.val:
                    current.next = list1
                    current = current.next
                    list1 = list1.next
                else:
                    current.next = list2
                    current = current.next
                    list2 = list2.next
                
        
        return sortedHead.next